src/stream-tcp-reassemble.c

```diff
@@ -1767,6 +1767,13 @@ int StreamTcpReassembleHandleSegment(ThreadVars *tv, TcpReassemblyThreadCtx *ra_
                 ssn, stream, p->payload_len,
                 (stream->flags & STREAMTCP_STREAM_FLAG_NOREASSEMBLY) ? "true" : "false");
 
+        /* In the case of bypass Suricata has to honor the inspection for all packets
+         * before stream depth so we need to check the depth.By adding the stream depth
+         * reached flag, it will trigger the bypass. */
+        if (StreamTcpBypassEnabled()) {
+            StreamTcpReassembleCheckDepth(ssn, stream, TCP_GET_SEQ(p), p->payload_len);
```

Comment:
I'm a bit confused about this. The last data packet (p->payload_len > 0) should have been checked, so depth flag should have been set then. The exception would be if NOREASSEMBLY flag is set, but perhaps the bypass needs to consider that flag instead?